File: AirTrafficControler/src/FlightCoordination/OptimizingThread.java
package FlightCoordination;

import java.util.concurrent.Semaphore;

public class OptimizingThread extends Thread{
	Integer wfs; 
	String ctrl;
	Semaphore sem; 
	Placement p;
	public OptimizingThread(Integer wfs, String ctrl, Semaphore sem, Placement p){
		super("OptimizingThread_");
		this.wfs  = wfs;
		this.ctrl = ctrl;
		this.sem  = sem;
		this.p    = p;
	}
	public void run(){
		try {
			sem.acquire();
				Utils.optimizePlacement(wfs,ctrl,p);
			sem.release();
		} catch (InterruptedException e) {
			e.printStackTrace();
		} catch (AssertionError e) {
			e.printStackTrace();
		}
	}
    public Placement getP(){
        return this.p;
    }
	public String getPid(){
		return p.getPid();
	}
	public double getExp(){
		return p.getFp().getExpectedValue();
	}
}

File: AirTrafficController_Python/OptimizingThread.py
import threading
import ParameterCore
import Utils
import Placement

class OptimizingThread:
    def __init__(self, wfs, ctrl, sem, p):
        self.wfs    = wfs
        self.ctrl   = ctrl
        self.sem    = sem
        self.p      = p

    def run(self):
        try:
            self.sem.acquire()
            Utils.Utils().optimizePlacement(self.wfs, self.ctrl, self.p)
            self.sem.release()
        except Exception as e:
            print(e)

    def getP(self):
        return self.p

    def getPid(self):
        return self.p.getPid()

    def getExp(self):
        return self.p.getFp().getExpectedValue()




